dot/core/service_test.go

```diff
@@ -117,6 +117,54 @@ func TestAnnounceBlock(t *testing.T) {
 	net.AssertCalled(t, "GossipMessage", expected)
 }
 
+func TestService_InsertKey(t *testing.T) {
+	ks := keystore.NewGlobalKeystore()
+
+	cfg := &Config{
+		Keystore: ks,
+	}
+	s := NewTestService(t, cfg)
+
+	kr, err := keystore.NewSr25519Keyring()
+	require.NoError(t, err)
+
+	testCases := []struct {
+		description  string
+		keystoreType string
+		err          error
+	}{
+		{
+			description:  "Test that insertKey fails when keystore type is invalid ",
+			keystoreType: "some-invalid-type",
+			err:          errors.New("invalid keystore name"),
+		},
+		{
+			description:  "Test that insertKey fails when keystore type is valid but inappropriate",
+			keystoreType: "gran",
+			err:          errors.New("invalid keystore name"),
+		},
+		{
+			description:  "Test that insertKey succeeds when keystore type is valid and appropriate ",
+			keystoreType: "acco",
+			err:          nil,
+		},
+	}
+
+	for _, c := range testCases {
+		c := c
+		t.Run(c.description, func(t *testing.T) {
+			err := s.InsertKey(kr.Alice(), c.keystoreType)
```

Comment:
Can you run this in parallel? (and also the parent test)

```suggestion
			t.Parallel()

			err := s.InsertKey(kr.Alice(), c.keystoreType)
```

If not, there is no point pinning `c := c` 😉

Reply:
removed `c := c`, that way I don't have to worry if these tests can run in parallel, these seem like light operations anyway.

Reply:
We need to run tests in parallel to find data races, it's not really for performance.

Reply:
done in the latest commit